pageserver/src/tenant/metadata.rs

```diff
@@ -232,6 +232,28 @@ impl TimelineMetadata {
     }
 }
 
+impl<'de> Deserialize<'de> for TimelineMetadata {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let bytes = Vec::<u8>::deserialize(deserializer)?;
```

Comment:
Yeah, I considered it -- I was satisfied with this because it's not a regression (what we previously did was deserialize to a Vec and then later deserialize the bytes from the Vec)